### PR TITLE
dataconnect(test): add sum partition and optional variable Arbs

### PR DESCRIPTION
**Pull request by dconeybe**
This PR introduces several property-based testing utilities to `firebase-dataconnect`, including a Kotest `Arb` for integer sum partitioning and generators for `OptionalVariable` types.

There should be no noticeable side effects to customers; this is internal test utility code.

### Highlights

* **Sum Partition Arbitrary**: Added `SumPartitionArb`, a Kotest `Arb` that generates random partitions of a non-negative integer sum into a fixed number of non-negative summands, including edge cases for zeroes, ascending order, and descending order.
* **OptionalVariable Generators**: Added `optionalVariable` and `nullableOptionalVariable` arbitrary generators to `DataConnectArb` to generate `OptionalVariable` instances with configurable probabilities for undefined and null states.
* **Kotest Test Helpers**: Added `PropTestConfig.withEdgeConfigEdgeCasesOnly()` for running tests solely against generated edge cases, and registered custom printer formatting for Apache Commons `SignificanceResult`.
* **Testing & Distribution Verification**: Added comprehensive unit and property-based test suites, including Chi-Square goodness-of-fit tests to verify uniform distributions of generated edge case types and probabilities.

<details>

<summary><b>Changelog</b></summary>

* **KotestTestutilPrinters.kt**
  * Registered custom printer support for Apache Commons `SignificanceResult`.
* **PropTestConfigExts.kt**
  * Added `withEdgeConfigEdgeCasesOnly()` extension on `PropTestConfig`.
* **SumPartitionArb.kt**
  * New Kotest `Arb` implementation for partitioning an integer sum into a fixed count of non-negative summands.
* **arbs.kt**
  * Added `optionalVariable` and `nullableOptionalVariable` arbitrary generator functions to `DataConnectArb`.
* **SumPartitionArbUnitTest.kt**
  * Added unit and property-based tests for `SumPartitionArb`.
* **arbsUnitTest.kt**
  * Added unit and property-based tests for `optionalVariable` and `nullableOptionalVariable`.
</details>

### CODE REVIEW

File: firebase-dataconnect/testutil/src/main/kotlin/com/google/firebase/dataconnect/testutil/property/arbitrary/SumPartitionArb.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    }
+  }
+
+  private val edgeCaseZeroesCountArb: Arb<Int> = run {
+    if (sum == 0) {
+      arbitrary { throw IllegalStateException("internal error h5zagzq8g4: should never get here") }
+    } else {
+      Arb.int(1 until summandCount)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firebase-dataconnect/testutil/src/main/kotlin/com/google/firebase/dataconnect/testutil/property/arbitrary/SumPartitionArb.kt:62]: Ignoring. I'm happy with this explicit code rather than introducing a mutex that is inherent with lazy

File: firebase-dataconnect/testutil/src/main/kotlin/com/google/firebase/dataconnect/testutil/property/arbitrary/arbs.kt [hunk 2]
```diff
@@ -329,6 +330,57 @@ object DataConnectArb {
...
+    nullableProbability: Double,
+  ): Arb<OptionalVariable<T?>> {
+    require(undefinedProbability in 0.0..1.0) {
+      "invalid undefinedProbability: ${undefinedProbability.print().value}"
+    }
+    require(nullableProbability in 0.0..1.0) {
+      "invalid nullableProbability: ${nullableProbability.print().value}"
+    }
+    val probabilitiesSum = undefinedProbability + nullableProbability
+    require(probabilitiesSum <= 1.0) {
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firebase-dataconnect/testutil/src/main/kotlin/com/google/firebase/dataconnect/testutil/property/arbitrary/arbs.kt:363]: Ignoring. The comment appears to be incorrect. First, I tested adding `0.56+0.44` and it is equal exactly to `1.0`. Second, I asked Antigravity to do some intense research and it concluded

> Surprisingly, if you are looking for two positive numbers (`a,b > 0`) that mathematically sum to `1.0`, no such pair exists in IEEE 754 double precision arithmetic (Java's double). The sum of any two positive numbers that mathematically equal `1.0` will always evaluate to `≤ 1.0` (either exactly `1.0` or `0.9999999999999999`).

**Review by yuchenshi — APPROVED** [review]
rubberstamp

**Review by stephenarosaj — APPROVED** [review]
LGTM
